tests/integration/backward_compatible/db_test.py

```diff
@@ -0,0 +1,123 @@
+from typing import List
+
+from hazelcast import HazelcastClient
+from hazelcast.config import Config
+from hazelcast.db import connect, Connection
```

Comment:
Moved these tests out out of backward compatibility.

Reply:
When I think more about it, the backward compatibility tests are for testing the compatibility of the server. And, this is basically a client-only feature on top of the backward compatible SQL feature. So, I think it might be OK to leave it like this. But, we still need to add server version checks to these tests to make sure that they will run correctly in client backward compatibility tests

Reply:
The server version is already checked, since the tests inherit "SqlTestBase"